src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -300,11 +304,29 @@ class PeerActor(
       def update(ns: Seq[BigInt]) = {
         val maxBlockNumber = ns.fold(0: BigInt) { case (a, b) => if (a > b) a else b }
         if (maxBlockNumber > currentMaxBlockNumber) {
-          context become new HandshakedHandler(rlpxConnection, initialStatus, maxBlockNumber).receive
+          context become new HandshakedHandler(rlpxConnection, initialStatus, maxBlockNumber, blockBroadcastActor).receive
         }
       }
     }
 
+    def handleBlockBroadcastActorTerminated: Receive = {
+      case Terminated(peer) if blockBroadcastActor.exists(_.compareTo(peer) == 0) =>
```

Comment:
We are also watching the termination of the `rlpxConnection` so this is done to differentiate the `Terminated` messages got in both cases. I realized that we weren't doing that on `handleTerminated` so I'll add that check.